src/Data/Fin/Properties.agda

```diff
@@ -761,6 +767,23 @@ pinch-mono-≤ 0F      {suc j} {suc k} (s≤s j≤k) = j≤k
 pinch-mono-≤ (suc i) {0F}    {k}     0≤n       = z≤n
 pinch-mono-≤ (suc i) {suc j} {suc k} (s≤s j≤k) = s≤s (pinch-mono-≤ i j≤k)
 
+pinch-injective :
+  {k : ℕ} {x : Fin k} {y z : Fin (ℕ.suc k)} →
+  suc x ≢ y → suc x ≢ z → pinch x y ≡ pinch x z → y ≡ z
+pinch-injective {y = zero} {zero} f g p = refl
+pinch-injective {x = zero} {zero} {suc z} f g p =
+  contradiction (cong suc p) g
+pinch-injective {x = zero} {suc y} {zero} f g p =
+  contradiction (cong suc (sym p)) f
+pinch-injective {x = zero} {suc y} {suc z} f g p =
+  cong suc p
+pinch-injective {x = suc x} {suc y} {suc z} f g p =
+  cong suc
+    ( pinch-injective
+      ( λ q → f (cong suc q))
```

Comment:
Can use function composition:
https://github.com/agda/agda-stdlib/blob/1f87ca2605635bbe05799e322d9ec0c43531873f/src/Function/Base.agda#L52

to rewrite these as `f ∘ cong suc` which should mean that the case's proof fits on 1 line?